app/public/src/pages/lobby.tsx

```diff
@@ -240,25 +240,26 @@ export async function joinLobbyRoom(
         dispatch(logIn(user))
         try {
           const token = await user.getIdToken()
-
-          let lobby: Room<ICustomLobbyState> | undefined
-          const reconnectToken: string = localStore.get(LocalStoreKeys.RECONNECTION_TOKEN)
+          let lobby: Room<ICustomLobbyState> | undefined = store.getState().network.lobby
+          if (lobby?.connection.isOpen) {
+            localStore.set(LocalStoreKeys.RECONNECTION_LOBBY, lobby.reconnectionToken, 30)
+            lobby.onLeave.clear()
+            await lobby.leave(false)
```

Comment:
What's the point of leaving the lobby room if you reconnect immediately to it a few lines after ?

Reply:
Couldn't this be simplified to:
```
 firebase.auth().onAuthStateChanged(async (user) => {
      if (user) {
        dispatch(logIn(user))
        try {
          const token = await user.getIdToken()
          let room: Room<ICustomLobbyState> | undefined = store.getState().network.lobby
          const reconnectToken: string = localStore.get(LocalStoreKeys.RECONNECTION_LOBBY)
          if ((!room || !room?.connection.isOpen) && reconnectToken) {
            try {
              room = await client.reconnect(reconnectToken)
            } catch (error) {
              localStore.delete(LocalStoreKeys.RECONNECTION_LOBBY)
            }
          }
          if(!room) {
            room = await client.join("lobby", {
              idToken: token
            })
            // store key for 5 minutes, but can still only rejoin within 30 seconds of leaving
            localStore.set(LocalStoreKeys.RECONNECTION_LOBBY, room.reconnectionToken, 60 * 5)
          }
```

Reply:
then what's the point to use a reconnection token in the first place if we need to force leaving and rejoining

Reply:
If i count correctly, with the prejoin from auth page, that's joining/leaving the lobby room 3 times in a row just for one login